pkg/operation/care/constraints.go

```diff
@@ -113,28 +113,21 @@ func (c *Constraint) constraintsChecks(
 
 		return []gardencorev1beta1.Condition{hibernationPossibleConstraint, maintenancePreconditionsSatisfiedConstraint}
 	}
-
 	if !apiServerRunning {
 		// don't check constraints if API server has already been deleted or has not been created yet
 		return shootControlPlaneNotRunningConstraints(hibernationPossibleConstraint, maintenancePreconditionsSatisfiedConstraint)
 	}
+	c.shootClient = shootClient.Client()
 
-	c.shootClient = client.Client()
-
-	var newHibernationConstraint, newMaintenancePreconditionsSatisfiedConstraint *gardencorev1beta1.Condition
-
-	status, reason, message, errorCodes, err := c.CheckForProblematicWebhooks(ctx)
-	if err == nil {
-		updatedHibernationCondition := gardencorev1beta1helper.UpdatedCondition(hibernationPossibleConstraint, status, reason, message, errorCodes...)
-		newHibernationConstraint = &updatedHibernationCondition
-
-		updatedMaintenanceCondition := gardencorev1beta1helper.UpdatedCondition(maintenancePreconditionsSatisfiedConstraint, status, reason, message, errorCodes...)
-		newMaintenancePreconditionsSatisfiedConstraint = &updatedMaintenanceCondition
+	status, reason, message, errorCodes, err = c.CheckForProblematicWebhooks(ctx)
+	if err != nil {
+		hibernationPossibleConstraint = NewConditionOrError(hibernationPossibleConstraint, nil, err)
+		maintenancePreconditionsSatisfiedConstraint = NewConditionOrError(maintenancePreconditionsSatisfiedConstraint, nil, err)
```

Comment:
Directly calling `gardencorev1beta1helper.UpdatedConditionUnknownError` would be even more readable :)